pkg/cloudprovider/provider/vsphere/helper.go

```diff
@@ -427,3 +433,33 @@ func getDatastoreFromVM(ctx context.Context, session *Session, vmRef *object.Vir
 	}
 	return session.Finder.Datastore(ctx, datastorePathObj.Datastore)
 }
+
+func resolveResourcePoolRef(ctx context.Context, config *Config, session *Session, vm *object.VirtualMachine) (*types.ManagedObjectReference, error) {
+	var targetResourcePool *object.ResourcePool
+	var isTemplate, err = vm.IsTemplate(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get vm template property: %v", err)
+	}
+	if config.ResourcePool != "" {
+		targetResourcePool, err = session.Finder.ResourcePool(ctx, config.ResourcePool)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get target resourcepool: %v", err)
+		}
+	} else if isTemplate {
+		var targetCluster *object.ClusterComputeResource
+		targetCluster, err = session.Finder.ClusterComputeResource(ctx, config.Cluster)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get target cluster: %v", err)
+		}
+		targetResourcePool, err = targetCluster.ResourcePool(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get target resourcepool from cluster: %v", err)
+		}
+	} else {
```

Comment:
this can be avoided, we could just not set  the resourcePool and leave it handle the defaulting as we had before

Reply:
okay I changed that now to return nil not sure if that is the correct way of doing it but it correctly worked in my testing and is using the pool of the source VM accordingly